tests/unit/helpers/QuexmlPDFTest.php

```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    /**
+     * Test format with reformat function disabled.
+     */
+    public function testKeepDbDateValues()
+    {
+        // Disable the setting
+        \Yii::app()->setConfig('quexmlkeepsurveydateformat',false);
+
+        $xpath = $this->_getXPath();
+
+        // Test question with date and time
+        $element = $xpath->query("//response[@varName='q1']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"2020-05-01 15:50:00","Unexpected value in question with date and time");
+
+        // Test question with time only
+        $element = $xpath->query("//response[@varName='q2']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"1970-01-01 15:55:00","Unexpected value in question with time only");
+
+        // Test question with date and time
+        $element = $xpath->query("//response[@varName='q3']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"2020-05-02 00:00:00","Unexpected value in question with date only");
+    }
+
+    /**
+     * Test format with reformat function enabled.
+     */
+    public function testKeepSurveyDateFormat()
+    {
+        // Enable the setting
+        \Yii::app()->setConfig('quexmlkeepsurveydateformat',true);
+
+        $xpath = $this->_getXPath();
+
+        // Test question with date and time
+        $element = $xpath->query("//response[@varName='q1']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"05/01/2020 15:50","Unexpected value in question with date and time");
+
+        // Test question with time only
+        $element = $xpath->query("//response[@varName='q2']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"15:55","Unexpected value in question with time only");
+
+        // Test question with date and time
+        $element = $xpath->query("//response[@varName='q3']")[0];
+        $value = $element->getAttribute("defaultValue");
+        $this->assertEquals($value,"2020-05-02","Unexpected value in question with date only");
+    }
+
+    /**
+     * Test XML generation with function disabled.
+     */
+    public function testDoNotUseQuestionTitle()
+    {
+        // Disable the setting
+        \Yii::app()->setConfig('quexmlusequestiontitleasid',false);
+
+        $id = $this->_getQuestionIdentifier();
+        $this->assertEquals($id,"A1.","Unexpected identifier for question 1.");
+
+    }
+
+    /**
+     * Test XML generation with function enabled.
+     */
+    public function testUseQuestionTitle()
+    {
+        // Enable the setting
+        \Yii::app()->setConfig('quexmlusequestiontitleasid',true);
+
+        $id = $this->_getQuestionIdentifier();
+        $this->assertEquals($id,"q1.","Unexpected identifier for question 1.");
+
+    }
+
+    protected function _getQuestionIdentifier()
```

Comment:
PSR-12, don't use underscore in method names.

Reply:
Fixed